### PR TITLE
Adding bamUtil

### PR DESCRIPTION
**Pull request by rleonid**
This is my first time creating a conda recipe, so I would encourage a thorough review.

### CODE REVIEW

**Comment by bgruening** [comment]
@rleonid thanks for your contribution. Can you add explicit compilers, like in this example: https://github.com/bioconda/bioconda-recipes/blob/0b5dbf8fefc34d7af4cbb8027e41c15516022008/recipes/cancerit-allelecount/meta.yaml

**Comment by rleonid** [comment]
@bgruening Can you point me to some documentation about how these compilers are setup? Are these just the default conda-recipes? I've built the library with `gcc` on linux, but the "gcc" on osx that is `clang` in disguise doesn't compile (will require a patch with the dual stdlibs).

**Comment by bgruening** [comment]
Sorry, no link to documentation. I can only point to other recipes. But can you try to add llvm # [osx] as osx compiler?

**Comment by rleonid** [comment]
Added.

**Comment by bgruening** [comment]
This looks good now. Can you please set the build number to 0 `number: 0`. This is the first build and it should start with 0.

Thanks!

File: recipes/bamutil/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - gcc # [linux]
+    - llvm  # [osx]
+  run:
+
```
Comment (johanneskoester) [inline recipes/bamutil/meta.yaml]: Can you please add tests as well? The invocation of `<cmd> --help` would suffice.

File: recipes/bamutil/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  sha256: "f5ec8d5e98a3797742106c3413a4ab1622d8787e38b29b3df4cddb59d77efda5"
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - gcc # [linux]
+    - llvm  # [osx]
+  run:
```
Comment (johanneskoester) [inline recipes/bamutil/meta.yaml]: Above line can be removed.

Comment (rleonid) [inline recipes/bamutil/meta.yaml]: Addressed,

**Comment by schelhorn** [comment]
Builds fine for me, but the libStatGen dependency requires the makefile to be able to call git for cloning libStatGen. So there is an implicit dependency, perhaps that's why the Travis build failed?

**Comment by rleonid** [comment]
I don't think that it is a `git` issue but rather a conda build one. `bamUtil` tries to clone and build `libstatGen` into `../libStatGen`... is that an issue for the build directory used by `conda build`?

**Comment by chapmanb** [comment]
Leonid -- thanks so much for getting this recipe started. I finalized it by explicitly downloading libStatGen instead of relying on the clone outside the work directory and pushed the recipe to the main branch. Apologies, I tried to do the git gymnastics to push this directly into your branch and PR but ended up messing everything up so thought this was the quickest way to make your work available. Thanks again for doing this.

**Pull request closed without merging**